File: swift/flatten-array/Sources/FlattenArray/FlattenArray.swift
func flattenArray<T>(_ arr: [Any?]) -> [T] {
  var res: [T] = []
  
  for e in arr {
    if let e = e as? [Any] {
        res += flattenArray(e)
    } else if e != nil {
        res.append(e as! T)
    }
  }
    return res
}

File: kotlin/flatten-array/src/main/kotlin/FlattenArray.kt
object Flattener {
    fun flatten(source: Collection<Any?>): List<Any> {
        var res = mutableListOf<Any>()
        for (e in source) {
            if (e is List<Any?>) {
                for (el in flatten(e)) {
                    res.add(el)
                }
            } else if (e != null) {
                res.add(e)
            }
        }
        return res
    }
}
